### PR TITLE
add CP2K test

### PR DESCRIPTION
**Pull request by smoors**

### CODE REVIEW

**Comment by boegel** [comment]
Can you add a small README file that mentions where we got the input files, when they were obtained, from which version, what the SHA256 checksum is, etc.?

File: eessi/testsuite/tests/apps/cp2k/input/QS/H2O-128.inp [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+      CUTOFF 280
+      REL_CUTOFF 30
+    &END MGRID
+    &QS
+      EPS_DEFAULT 1.0E-12
+      EXTRAPOLATION_ORDER 3
+      WF_INTERPOLATION PS
+    &END QS
+    &SCF
+      # IGNORE_CONVERGENCE_FAILURE
```
Comment (boegel) [inline eessi/testsuite/tests/apps/cp2k/input/QS/H2O-128.inp:32]: make it clear why this is commented out?

```suggestion
      # IGNORE_CONVERGENCE_FAILURE  # only for CP2K >= 2024.1
```

`sed 's/### IGNORE_CONVERGENCE_FAILURE/ IGNORE_CONVERGENCE_FAILURE /g`

**Comment by laraPPr** [comment]
Tested on out local cluster in Gent, result:
```
[     FAIL ] ( 1/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=1_cpn_4_nodes /f5075236 @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_f5075236'
[     FAIL ] ( 2/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=4_cores /5da5d139 @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_5da5d139'
[     FAIL ] ( 3/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=1_core /6c2cca0a @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_6c2cca0a'
[     FAIL ] ( 4/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=2_cores /a5cd2ae1 @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_a5cd2ae1'
[     FAIL ] ( 5/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=1_cpn_2_nodes /8cfc4ef7 @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_8cfc4ef7'
[     FAIL ] ( 6/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=1_core /bfb78618 @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_bfb78618'
[       OK ] ( 7/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=1_8_node /738c12b8 @doduo:doduo+default
P: time: 36.912 s (r:0, l:None, u:None)
[     FAIL ] ( 8/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=1_8_node /a6fbe794 @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_a6fbe794'
[       OK ] ( 9/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=4_cores /d0a864b6 @doduo:doduo+default
P: time: 146.342 s (r:0, l:None, u:None)
[       OK ] (10/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=1_cpn_4_nodes /6ccf849c @doduo:doduo+default
P: time: 172.019 s (r:0, l:None, u:None)
[       OK ] (11/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=2_cores /613b66ce @doduo:doduo+default
P: time: 190.065 s (r:0, l:None, u:None)
[       OK ] (12/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=1_cpn_2_nodes /eed9d55a @doduo:doduo+default
P: time: 207.19 s (r:0, l:None, u:None)
[       OK ] (13/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=1_8_node /f8738793 @doduo:doduo+default
P: time: 255.705 s (r:0, l:None, u:None)
[       OK ] (14/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=1_core /508f6ee8 @doduo:doduo+default
P: time: 348.871 s (r:0, l:None, u:None)

[       OK ] (15/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=4_cores /ba32b280 @doduo:doduo+default
P: time: 623.666 s (r:0, l:None, u:None)
[       OK ] (16/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=1_cpn_4_nodes /85211789 @doduo:doduo+default
P: time: 726.105 s (r:0, l:None, u:None)
[     FAIL ] (17/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=1_cpn_2_nodes /185f370e @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_185f370e'
[       OK ] (18/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=1_4_node /5e129d50 @doduo:doduo+default
P: time: 141.607 s (r:0, l:None, u:None)
[       OK ] (19/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=1_4_node /81403de6 @doduo:doduo+default
P: time: 22.382 s (r:0, l:None, u:None)
[     FAIL ] (20/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=2_cores /7b235c7a @doduo:doduo+default
==> test failed during 'sanity': test staged in '/kyukon/scratch/gent/461/vsc46128/EESSI/test-suite/stage/doduo/doduo/default/EESSI_CP2K_7b235c7a'
[       OK ] (21/39) EESSI_CP2K %benchmark_info=QS/H2O-128 %module_name=CP2K/2023.1-foss-2023a %scale=1_2_node /0e489b46 @doduo:doduo+default
P: time: 107.325 s (r:0, l:None, u:None)
[       OK ] (22/39) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/2023.1-foss-2023a %scale=1_2_node /0bd9f24b @doduo:doduo+default
P: time: 17.458 s (r:0, l:None, u:None)
[       OK ] (23/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=1_2_node /b23e06dc @doduo:doduo+default
P: time: 1453.179 s (r:0, l:None, u:None)
[       OK ] (24/39) EESSI_CP2K %benchmark_info=QS/H2O-512 %module_name=CP2K/2023.1-foss-2023a %scale=1_4_node /b35c39de @doduo:doduo+default
P: time: 1988.409 s (r:0, l:None, u:None)
```

A majority of the `QS/H2O-512` and some of the `QS/H2O-128` failed. (They seem to be causes by mpirun. Here you can see an example:
```
2024-05-21 15:51:25,936 ERROR      RunAsyncMPI     MainThread  _post_exitcode: problem occured with cmd ['mpirun', '-machinefile', '/user/gent/461/vsc46128/.mympirun_p24k95/63206290_20240521_155104/nodes', '--mca', 'pml', 'ucx', '--mca', 'btl', '^uct,ofi', '--mca', 'mtl', '^ofi', '--mca', 'orte_keep_fqdn_hostnames', '1', '--mca', 'MKL_NUM_THREADS', '1', '--mca', 'MODULEPATH', '/apps/gent/RHEL8/zen2-ib/modules/all:/data/gent/vo/000/gvo00002/vsc46128/easybuild/RHEL8/zen2-ib/modules/all:/etc/modulefiles/vsc', '--mca', 'LOADEDMODULES', 'env/vsc/doduo:env/slurm/doduo:env/software/doduo:cluster/doduo:vsc-mympirun/5.4.0:GCCcore/12.3.0:zlib/1.2.13-GCCcore-12.3.0:binutils/2.40-GCCcore-12.3.0:GCC/12.3.0:numactl/2.0.16-GCCcore-12.3.0:XZ/5.4.2-GCCcore-12.3.0:libxml2/2.11.4-GCCcore-12.3.0:libpciaccess/0.17-GCCcore-12.3.0:hwloc/2.9.1-GCCcore-12.3.0:OpenSSL/1.1:libevent/2.1.12-GCCcore-12.3.0:UCX/1.14.1-GCCcore-12.3.0:PMIx/4.2.4-GCCcore-12.3.0:UCC/1.2.0-GCCcore-12.3.0:OpenMPI/4.1.5-GCC-12.3.0:OpenBLAS/0.3.23-GCC-12.3.0:FlexiBLAS/3.3.1-GCC-12.3.0:FFTW/3.3.10-GCC-12.3.0:gompi/2023a:FFTW.MPI/3.3.10-gompi-2023a:ScaLAPACK/2.2.0-gompi-2023a-fb:foss/2023a:Libint/2.7.2-GCC-12.3.0-lmax-6-cp2k:libxc/6.2.2-GCC-12.3.0:libxsmm/1.17-GCC-12.3.0:libvori/220621-GCCcore-12.3.0:GSL/2.7-GCC-12.3.0:bzip2/1.0.8-GCCcore-12.3.0:ncurses/6.4-GCCcore-12.3.0:libreadline/8.2-GCCcore-12.3.0:Tcl/8.6.13-GCCcore-12.3.0:SQLite/3.42.0-GCCcore-12.3.0:libffi/3.4.4-GCCcore-12.3.0:Python/3.11.3-GCCcore-12.3.0:gfbf/2023a:cffi/1.15.1-GCCcore-12.3.0:cryptography/41.0.1-GCCcore-12.3.0:virtualenv/20.23.1-GCCcore-12.3.0:Python-bundle-PyPI/2023.06-GCCcore-12.3.0:pybind11/2.11.1-GCCcore-12.3.0:SciPy-bundle/2023.07-gfbf-2023a:gzip/1.12-GCCcore-12.3.0:lz4/1.9.4-GCCcore-12.3.0:zstd/1.5.5-GCCcore-12.3.0:ICU/73.2-GCCcore-12.3.0:Boost/1.82.0-GCC-12.3.0:PLUMED/2.9.0-foss-2023a:CP2K/2023.1-foss-2023a', '--mca', 'MODULESHOME', '/usr/share/lmod/lmod', '-np', '4', '-x', 'LD_LIBRARY_PATH', '-x', 'PATH', '-x', 'PYTHONPATH', '-x', 'OMP_PROC_BIND', '-x', 'OMP_NUM_THREADS', '--map-by', 'ppr:1:node:PE=1:SPAN:NOOVERSUBSCRIBE', 'cp2k.popt', '-i', 'H2O-512.inp']: (shellcmd ['mpirun', '-machinefile', '/user/gent/461/vsc46128/.mympirun_p24k95/63206290_20240521_155104/nodes', '--mca', 'pml', 'ucx', '--mca', 'btl', '^uct,ofi', '--mca', 'mtl', '^ofi', '--mca', 'orte_keep_fqdn_hostnames', '1', '--mca', 'MKL_NUM_THREADS', '1', '--mca', 'MODULEPATH', '/apps/gent/RHEL8/zen2-ib/modules/all:/data/gent/vo/000/gvo00002/vsc46128/easybuild/RHEL8/zen2-ib/modules/all:/etc/modulefiles/vsc', '--mca', 'LOADEDMODULES', 'env/vsc/doduo:env/slurm/doduo:env/software/doduo:cluster/doduo:vsc-mympirun/5.4.0:GCCcore/12.3.0:zlib/1.2.13-GCCcore-12.3.0:binutils/2.40-GCCcore-12.3.0:GCC/12.3.0:numactl/2.0.16-GCCcore-12.3.0:XZ/5.4.2-GCCcore-12.3.0:libxml2/2.11.4-GCCcore-12.3.0:libpciaccess/0.17-GCCcore-12.3.0:hwloc/2.9.1-GCCcore-12.3.0:OpenSSL/1.1:libevent/2.1.12-GCCcore-12.3.0:UCX/1.14.1-GCCcore-12.3.0:PMIx/4.2.4-GCCcore-12.3.0:UCC/1.2.0-GCCcore-12.3.0:OpenMPI/4.1.5-GCC-12.3.0:OpenBLAS/0.3.23-GCC-12.3.0:FlexiBLAS/3.3.1-GCC-12.3.0:FFTW/3.3.10-GCC-12.3.0:gompi/2023a:FFTW.MPI/3.3.10-gompi-2023a:ScaLAPACK/2.2.0-gompi-2023a-fb:foss/2023a:Libint/2.7.2-GCC-12.3.0-lmax-6-cp2k:libxc/6.2.2-GCC-12.3.0:libxsmm/1.17-GCC-12.3.0:libvori/220621-GCCcore-12.3.0:GSL/2.7-GCC-12.3.0:bzip2/1.0.8-GCCcore-12.3.0:ncurses/6.4-GCCcore-12.3.0:libreadline/8.2-GCCcore-12.3.0:Tcl/8.6.13-GCCcore-12.3.0:SQLite/3.42.0-GCCcore-12.3.0:libffi/3.4.4-GCCcore-12.3.0:Python/3.11.3-GCCcore-12.3.0:gfbf/2023a:cffi/1.15.1-GCCcore-12.3.0:cryptography/41.0.1-GCCcore-12.3.0:virtualenv/20.23.1-GCCcore-12.3.0:Python-bundle-PyPI/2023.06-GCCcore-12.3.0:pybind11/2.11.1-GCCcore-12.3.0:SciPy-bundle/2023.07-gfbf-2023a:gzip/1.12-GCCcore-12.3.0:lz4/1.9.4-GCCcore-12.3.0:zstd/1.5.5-GCCcore-12.3.0:ICU/73.2-GCCcore-12.3.0:Boost/1.82.0-GCC-12.3.0:PLUMED/2.9.0-foss-2023a:CP2K/2023.1-foss-2023a', '--mca', 'MODULESHOME', '/usr/share/lmod/lmod', '-np', '4', '-x', 'LD_LIBRARY_PATH', '-x', 'PATH', '-x', 'PYTHONPATH', '-x', 'OMP_PROC_BIND', '-x', 'OMP_NUM_THREADS', '--map-by', 'ppr:1:node:PE=1:SPAN:NOOVERSUBSCRIBE', 'cp2k.popt', '-i', 'H2O-512.inp']) output  DBCSR| CPU Multiplication driver                                           XSMM (U)
 DBCSR| Multrec recursion limit                                              512 (U)
 DBCSR| Multiplication stack size                                           1000 (D)
 DBCSR| Maximum elements for images                                    UNLIMITED (U)
 DBCSR| Multiplicative factor virtual images                                   1 (U)
 DBCSR| Use multiplication densification                                       T (D)
 DBCSR| Multiplication size stacks                                             3 (U)
 DBCSR| Use memory pool for CPU allocation                                     F (U)
 DBCSR| Number of 3D layers                                               SINGLE (U)
 DBCSR| Use MPI memory allocation                                              F (U)
--- rfm_job.err ---
```

**Comment by smoors** [comment]
thanks for testing!

`QS/H2O-512` is rather big, needs quite some memory. seems likely that the failures are due to OOM, which correlates with the fact that it only runs successfully on the `1_2_node` and `1_4_node` scales.

**Comment by smoors** [comment]
here are some memory usage numbers for QS/H2O-512:
- 1_8_node (zen4 8 cores): 4.15G per rank = 33.2G total, before hitting time limit (1h, was at step number 9, 4.5 fs)
- 1_4_node (zen4 16 cores): 2.14G per rank = 34.24G total, finished in 37:10

so in this case the memory is mostly constant with a small amount added per rank

**Comment by smoors** [comment]
memory

**Comment by casparvl** [comment]
Nice, with this data and [this PR](https://github.com/Crivella/test-suite/pull/1) it should be quite straightforward to ask the right amount of memory.

**Comment by smoors** [comment]
image

this is for H2O-128
- 8 cores 5.13G
- 16 cores 6.13G
- 32 cores 8.05G

**Comment by casparvl** [comment]
https://github.com/EESSI/test-suite/pull/128 was merged, so you can now use the hook to request the required amount of memory :)

**Comment by smoors** [comment]
> #128 was merged, so you can now use the hook to request the required amount of memory :)

done in https://github.com/EESSI/test-suite/pull/133/commits/ebe7257bf338e0d2476ba652bc11bdfed6853d31

**Comment by casparvl** [comment]
Hm... something is still fishy with the memory request. On Snellius, I get OOM failures for the `2_cores` and `4_cores` cases. This is the job script for the `4_cores` case:

```
#!/bin/bash
#SBATCH --job-name="rfm_EESSI_CP2K_5da5d139"
#SBATCH --ntasks=4
#SBATCH --ntasks-per-node=4
#SBATCH --cpus-per-task=1
#SBATCH --output=rfm_job.out
#SBATCH --error=rfm_job.err
#SBATCH --time=1:0:0
#SBATCH -p rome
#SBATCH --export=None
#SBATCH --mem=34571M
source /cvmfs/software.eessi.io/versions/2023.06/init/bash
module load CP2K/2023.1-foss-2023a
export OMP_NUM_THREADS=1
cp /gpfs/home4/casparl/EESSI/test-suite/eessi/testsuite/tests/apps/cp2k/input/QS/H2O-512.inp ./
mpirun -np 4 cp2k.popt -i H2O-512.inp
``` 
On Vega, practically everything fails, not sure yet if that's OOM as well, some of it just seems to be... I dunno... I/O stuff. Probably more an issue with the system than with your test.

On Karolina, everything passes (so far).

Any idea why that Snellius job went OOM? Could it be that the test requires more memory than you originally anticipated at low core counts?

**Comment by smoors** [comment]
yes, the QS/H2O-512 benchmark should be run on at least 16 cores to finish in < 1 hour.
i’ve now added a `skip_if` hook that skips tests running with less than 16 cores, but even then you may hit the time limit if your cpus are not fast enough.
(with CP2K/2023.1-foss-2023a and 16 cores on a Genoa-X took ~34 minutes)

it's indeed possible that i underestimated the required memory for low core counts due to nonlinear behavior, but with the `skip_if` that should no longer be an issue.

**Comment by laraPPr** [comment]
Failures on Hortense:
```
{'Failed': 
  {
    'module load CP2K/8.2-foss-2021a': 6, 
    'module load CP2K/9.1-foss-2022a': 3, 
    'module load CP2K/2022.1-foss-2022a': 3, 
    'module load CP2K/5.1-foss-2020b': 35, 
    'module load CP2K/7.1-foss-2020b': 35, 
    'module load CP2K/8.1-foss-2020b': 35, 
    'module load CP2K/2023.1-foss-2022b': 35
  }, 
'passed': 
  {
    'module load CP2K/8.2-foss-2021a': 29, 
    'module load CP2K/9.1-foss-2022a': 32, 
    'module load CP2K/2022.1-foss-2022a': 32}}
```
I will focus on why some of `8.2`, `9.1` and `2022.1` are failing

**Comment by laraPPr** [comment]
`module load CP2K/5.1-foss-2020b` failed because of unknow keyword

```
Possible matches for unknown keyword 

 EPS_CHECK_DIAG

   keyword EPS_CN in section %__ROOT__%FORCE_EVAL%DFT%KG_METHOD%ENERGY_CORRECTION%XC%VDW_POTENTIAL%PAIR_POTENTIAL score:  6
   keyword EPS_CN in section %__ROOT__%FORCE_EVAL%DFT%TDDFPT%XC%VDW_POTENTIAL%PAIR_POTENTIAL score:  6
   keyword EPS_CN in section %__ROOT__%FORCE_EVAL%DFT%XC%VDW_POTENTIAL%PAIR_POTENTIAL score:  6
   keyword EPS_CN in section %__ROOT__%FORCE_EVAL%PROPERTIES%LINRES%EPR%PRINT%G_TENSOR%XC%VDW_POTENTIAL%PAIR_POTENTIAL score:  6
   keyword EPS_CN in section %__ROOT__%ATOM%METHOD%XC%VDW_POTENTIAL%PAIR_POTENTIAL score:  6

 *******************************************************************************
 *   ___                                                                       *
 *  /   \                                                                      *
 * [ABORT]                                                                     *
 *  \___/        found an unknown keyword EPS_CHECK_DIAG in section GLOBAL     *
 *    |                                                                        *
 *  O/|                                                                        *
 * /| |                                                                        *
 * / \                                               input/input_parsing.F:246 *
 *******************************************************************************


 ===== Routine Calling Stack ===== 
```

**Comment by laraPPr** [comment]
`CP2K/2023.1-foss-2022b` failed with the following error:
```
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
```

**Comment by laraPPr** [comment]
For `CP2K/8.2-foss-2021a` 

The `16_nodes`  all timed out
The `1_core` ran so I think the sanity check failed but I can't find it in the reframe logs.
and a `1_node` test was killed:
```
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank 8 with PID 1270970 on node node5385.dodrio.os exited on signal 9 (Killed).
slurmstepd: error: Detected 1 oom_kill event in StepId=8327471.batch. Some of the step tasks have been OOM Killed.
```

**Comment by smoors** [comment]
> `CP2K/5.1-foss-2020b`, `CP2K/7.1-foss-2020b` and `CP2K/8.1-foss-2020b` failed because of unknow keyword

keyword `EPS_CHECK_DIAG` was probably added in more recent versions of CP2K.
this keyword is not absolutely required, but it's nice to have as a check for numerical accuracy.
i could remove this keyword for older versions or skip older versions altogether.
i vote for skipping them, as they are already quite old and are probably not going to be added to EESSI.

**Comment by smoors** [comment]
> For `CP2K/8.2-foss-2021a`
> 
> The `16_nodes` all timed out

yeah, 16 nodes is way too much for these tests, i'll skip those too. which was the highest node number that worked for each of the 3 tests?

> The `1_core` ran so I think the sanity check failed but I can't find it in the reframe logs.

which of the 3 tests was this exactly?

and a `1_node` test was killed:
> 
> ```
> --------------------------------------------------------------------------
> Primary job  terminated normally, but 1 process returned
> a non-zero exit code. Per user-direction, the job has been aborted.
> --------------------------------------------------------------------------
> --------------------------------------------------------------------------
> mpirun noticed that process rank 8 with PID 1270970 on node node5385.dodrio.os exited on signal 9 (Killed).
> slurmstepd: error: Detected 1 oom_kill event in StepId=8327471.batch. Some of the step tasks have been OOM Killed.
> ```
> 
> Also saw similar things for `CP2K/9.1-foss-2022a` and `CP2K/2022.1-foss-2022a`

also here, which test?

**Comment by laraPPr** [comment]
> > and a `1_node` test was killed:
> > ```
> > --------------------------------------------------------------------------
> > Primary job  terminated normally, but 1 process returned
> > a non-zero exit code. Per user-direction, the job has been aborted.
> > --------------------------------------------------------------------------
> > --------------------------------------------------------------------------
> > mpirun noticed that process rank 8 with PID 1270970 on node node5385.dodrio.os exited on signal 9 (Killed).
> > slurmstepd: error: Detected 1 oom_kill event in StepId=8327471.batch. Some of the step tasks have been OOM Killed.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also saw similar things for `CP2K/9.1-foss-2022a` and `CP2K/2022.1-foss-2022a`
> 
> also here, which test? the mem requirements were measured with CP2K2023.1, it's possible that other versions require more memory (e.g. due to mem leaks).

[     FAIL ] (207/273) EESSI_CP2K %benchmark_info=QS/H2O-32 %module_name=CP2K/8.2-foss-2021a %scale=1_node /04078396 @hortense:cpu_milan+default

**Comment by laraPPr** [comment]
> `CP2K/2023.1-foss-2022b` failed with the following error:
> 
> ```
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> cp2k.popt: error while loading shared libraries: libxcf03.so.12: cannot open shared object file: No such file or directory
> ```

This error only shows up when EESSI is initiated. When it is not the test runs correctly.

**Comment by casparvl** [comment]
> yeah, 16 nodes is way too much for these tests, i'll skip those too. which was the highest node number that worked for each of the 3 tests?

8 nodes worked for all 3 cases on Karolina and Snellius (on Vega, these are still in the queue). 

I'm a bit puzzled by the timings though. On Snellius, we have `CP2K/2023.1-foss-2023a`. There, even the 16 node cases completed just fine, with time to spare (`H2O-512` took 2233s on 16 nodes, the other two were around 800-820s on 16 nodes). On Karolina (`CP2K/9.1-foss-2022a`), the H2O-512 and H2O-32 use cases failed on 16 nodes, but the `H2O-128` succeeded in 1164 seconds. Isn't it weird that the medium-sized systems completes, but the others don't? :\

File: eessi/testsuite/hooks.py [hunk 2]
```diff
@@ -573,6 +574,15 @@ def set_compact_thread_binding(test: rfm.RegressionTest):
     log(f'Set environment variable KMP_AFFINITY to {test.env_vars["KMP_AFFINITY"]}')
 
 
+def set_omp_num_threads(test: rfm.RegressionTest):
+    """
+    Set default number of OpenMP threads equal to number of CPUs per task,
+    unless OMP_NUM_THREADS is already set
+    """
+    test.env_vars['OMP_NUM_THREADS'] = os.getenv('OMP_NUM_THREADS', test.num_cpus_per_task)
```
Comment (casparvl) [inline eessi/testsuite/hooks.py:582]: I don't think it makes sense to do this conditionally. What you are checking is if `OMP_NUM_THREADS` is set on the node where the `reframe` command runs (typically: login node). What you are setting is an `export OMP_NUM_THREADS=<something>` in the job script, which will cause it to be set on the batch node running the job.

Those are typically environments that have nothing to do with each other: we advise (in our docs) to configure reframe to submit with `--export=None` (https://www.eessi.io/docs/test-suite/ReFrame-configuration-file/) , precisely to make sure the submission environment doesn't affect the environment of the batch job.

I'd suggest just setting this in the test as `test.env_vars['OMP_NUM_THREADS'] = test.num_cpus_per_task`, unconditionally (pretty much similar to what we do for GROMACS, even if that is a bit more complex: https://github.com/EESSI/test-suite/blob/926c3e4859f2c9dbe0b9034202d317cdce331fba/eessi/testsuite/tests/apps/gromacs.py#L99)

**Comment by casparvl** [comment]
I've reran on Karolina. All 16 nodes test still time out - you were planning to remove that scale, but I think you forgot. I also had a time out on the `H20-512` on `1_8_node`. Now, I now that Karolina is clocked at a _relatively_ low clock frequency (I think it's 2.0 or 2.1 GHz or something). But it's not unheard off. I guess either we should increase the `time_limit` or increase the minimum core count for this problem size.

On Snellius, things are still running. I'll report back once that is one. 55 successes, 2 failures so far, but related to our system: one node fail, another a bit unclear but I suspect a filesystem mount that is inaccessible. (out of 78 total tests, rest is still running)

**Comment by smoors** [comment]
thanks for testing again.
i've removed the 16_nodes scale, and increased the time limit to 2 hours.
i think the QS/H2O-512 test is useful to run on 16 cores if your hardware is fast enough: on a zen4 machine it took just 30 minutes.

**Comment by casparvl** [comment]
On Snellius:
```
[  PASSED  ] Ran 66/78 test case(s) from 39 check(s) (0 failure(s), 12 skipped, 0 aborted)
```
On Karolina:
```
[  PASSED  ] Ran 33/39 test case(s) from 39 check(s) (0 failure(s), 6 skipped, 0 aborted)
```
Looks good to me. I'm going to merge this. If we need to tweak the memory limits further, we can always do so in a follow-up PR.

The failures that @laraPPr saw with very old versions of CP2K (due to that missing keyword), I don't think we should worry about it too much. If you want, you can make a follow-up PR that checks the version and skips if it's too old. But seeing as this is in first place a test suite for EESSI, and since EESSI does _not_ have these very old CP2K versions anyway... It's nice that the test suite _can_ be used to test local software environments, but it's not the primary goal. Thus, I think we're fine merging like this.

**Review by casparvl — APPROVED** [review]
Lgtm!

**Pull request merged**